File: backend/scripts/telegramUploadOnly.py
import os
import sys
from pathlib import Path
from pyrogram import Client
from dotenv import load_dotenv

env_path = Path(__file__).parent.parent.parent / '.env'
load_dotenv(env_path)

API_ID = os.getenv("TELEGRAM_API_ID")
API_HASH = os.getenv("TELEGRAM_API_HASH")
CHANNEL_ID_STR = os.getenv("TELEGRAM_CHANNEL_ID")

if not API_ID or not API_HASH or not CHANNEL_ID_STR:
    print("Erro: Variáveis do Telegram ausentes no .env (API_ID, API_HASH, CHANNEL_ID)")
    sys.exit(1)

CHANNEL_ID = int(CHANNEL_ID_STR)

def progress(current, total):
    print(f"\rUpload progresso: {current * 100 / total:.2f}%", end="")
    sys.stdout.flush()

def main():
    if len(sys.argv) < 2:
        print("Uso: python telegramUploadOnly.py <caminho_do_arquivo> [titulo_do_filme]")
        sys.exit(1)
        
    file_path = sys.argv[1]
    title = sys.argv[2] if len(sys.argv) > 2 else Path(file_path).stem

    if not os.path.exists(file_path):
        sys.exit(1)

    print(f"Iniciando upload de: {title} via Pyrogram...")
    
    script_dir = Path(__file__).parent.resolve()
    session_name = str(script_dir / "my_account")
    
    app = Client(session_name, api_id=int(API_ID), api_hash=API_HASH)
    
    with app:
        try:
            app.get_chat(CHANNEL_ID)
        except:
            for _ in app.get_dialogs():
                pass

        msg = app.send_video(
            chat_id=CHANNEL_ID,
            video=file_path,
            caption=f"**{title}**",
            supports_streaming=True,
            progress=progress
        )
        print(f"\nUpload concluído com sucesso! MESSAGE_ID: {msg.id}")

if __name__ == "__main__":
    main()

File: backend/scripts/telegramUploadOnly.js
import { TelegramClient } from "telegram";
import { StringSession } from "telegram/sessions/index.js";
import fs from "fs";
import 'dotenv/config';

const apiId = parseInt(process.env.TELEGRAM_API_ID);
const apiHash = process.env.TELEGRAM_API_HASH;
const sessionStr = process.env.TELEGRAM_SESSION;
const channelId = process.env.TELEGRAM_CHANNEL_ID;

if (!apiId || !apiHash || !sessionStr || !channelId) {
    console.error("Faltam variáveis no .env");
    process.exit(1);
}

const stringSession = new StringSession(sessionStr);

(async () => {
    const filePath = process.argv[2];
    if (!filePath || !fs.existsSync(filePath)) {
        console.error("Uso: node telegramUploadOnly.js /caminho/do/arquivo.mp4");
        process.exit(1);
    }

    console.log("Conectando ao Telegram...");
    const client = new TelegramClient(stringSession, apiId, apiHash, {
        connectionRetries: 10,
    });
    
    client.setLogLevel("none");
    await client.connect();
    console.log("Conectado! Iniciando upload de", filePath);

    let entityId = channelId;
    if (!entityId.startsWith('-100')) {
        entityId = '-100' + entityId.replace('-', '');
    }

    try {
        await client.sendFile(entityId, {
            file: filePath,
            workers: 1, // Usando 1 worker para garantir 100% de estabilidade no teste
            caption: `**Upload Manual**\n${filePath}`,
            parseMode: "markdown",
            progressCallback: (progress) => {
                process.stdout.write(`\rUpload progresso: ${(progress * 100).toFixed(2)}%`);
            }
        });
        
        process.stdout.write('\n');
        console.log("Upload concluído com sucesso pro Canal!");
    } catch (err) {
        console.error("Erro no upload:", err);
    } finally {
        process.exit(0);
    }
})();
